cmd/spiffe-helper/main.go

```diff
@@ -3,52 +3,55 @@ package main
 import (
 	"context"
 	"flag"
-	"fmt"
 	"os"
 	"os/signal"
 	"syscall"
 
 	"github.com/sirupsen/logrus"
 	"github.com/spiffe/spiffe-helper/cmd/spiffe-helper/config"
+	"github.com/spiffe/spiffe-helper/pkg/health"
 	"github.com/spiffe/spiffe-helper/pkg/sidecar"
 )
 
-const (
-	daemonModeFlagName = "daemon-mode"
-)
-
 func main() {
 	configFile := flag.String("config", "helper.conf", "<configFile> Configuration file path")
-	daemonModeFlag := flag.Bool(daemonModeFlagName, true, "Toggle running as a daemon to rotate X.509/JWT or just fetch and exit")
+	daemonModeFlag := flag.Bool(config.DaemonModeFlagName, true, "Toggle running as a daemon to rotate X.509/JWT or just fetch and exit")
 	flag.Parse()
 	log := logrus.WithField("system", "spiffe-helper")
 
-	if err := startSidecar(*configFile, *daemonModeFlag, log); err != nil {
+	hclConfig, err := config.ParseConfigFile(log, *configFile, *daemonModeFlag)
+	if err != nil {
+		log.WithError(err).Errorf("failed to parse configuration")
+		os.Exit(1)
+	}
+
+	if err := hclConfig.ValidateConfig(log); err != nil {
+		log.WithError(err).Errorf("invalid configuration")
+		os.Exit(1)
+	}
+
+	if err := startSidecar(hclConfig, log); err != nil {
 		log.WithError(err).Errorf("Error starting spiffe-helper")
 		os.Exit(1)
 	}
 
+	if err := health.StartHealthServer(hclConfig, log, spiffeSidecar); err != nil {
+		log.WithError(err).Errorf("Error starting spiffe-helper health check server")
+		os.Exit(1)
+	}
+
 	log.Infof("Exiting")
 	os.Exit(0)
 }
 
-func startSidecar(configFile string, daemonModeFlag bool, log logrus.FieldLogger) error {
+var spiffeSidecar *sidecar.Sidecar
```

Comment:
Any reason for this global variable?

Reply:
It's set by `startSidecar` and passed into `StartHealthServer`

Reply:
Changed to local variable.